# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Tenuring.cpp
# Commit: 28feda14fdec
# Full Hash: 28feda14fdec962ea68c26a0570ea69042a5ad27
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Add tagging for JSString* in AnyRef. r=yury
#   
#   Before this commit we would need to box a JSString* in WasmValueBox
#   when we received one from JS and converted to an AnyRef.
#   
# ==============================================================================

diff -r 85605041672d -r 28feda14fdec js/src/gc/Tenuring.cpp
--- a/js/src/gc/Tenuring.cpp	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/gc/Tenuring.cpp	Fri Aug 04 14:06:50 2023 +0000
@@ -183,6 +183,12 @@
       post = wasm::AnyRef::fromJSObject(*obj);
       break;
     }
+    case wasm::AnyRefKind::String: {
+      JSString* str = value.toJSString();
+      onStringEdge(&str, "string");
+      post = wasm::AnyRef::fromJSString(str);
+      break;
+    }
     case wasm::AnyRefKind::Null: {
       // This function must only be called for GC things.
       MOZ_CRASH();